### PR TITLE
Openai decorators

### PR DESCRIPTION
**Pull request by gavin-aguiar**
Added new decorators for OpenAI

New bindings:
- assistant_skill_trigger
- text_completion_input
- assistant_create_output
- assistant_query_input
- assistant_post_input
- embeddings_input
- semantic_search_input
- embeddings_store_output

### CODE REVIEW

File: azure/functions/decorators/function_app.py [hunk 1]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+                         model: Optional[str] = None,
+                         max_chunk_length: Optional[int] = 8 * 1024,
+                         max_overlap: Optional[int] = 128,
+                         data_type: Optional[
+                             Union[DataType, str]] = None,
+                         **kwargs) \
+            -> Callable[..., Any]:
+        """
+        The embeddings input decorator creates embeddings which will be used to
+        measure the readiness of text strings.
```
Comment (manvkaur) [inline azure/functions/decorators/function_app.py:2995]: nit: it should be `relatedness`

File: azure/functions/decorators/function_app.py [hunk 2]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+                              chat_model: Optional[OpenAIModels] = OpenAIModels.DefaultChatModel,  # NoQA
+                              system_prompt: Optional[str] = semantic_search_system_prompt,  # NoQA
+                              max_knowledge_count: Optional[int] = 1,
+                              data_type: Optional[
+                                  Union[DataType, str]] = None,
+                              **kwargs) \
+            -> Callable[..., Any]:
+        """
+        The embeddings input decorator creates embeddings which will be used to
+        measure the readiness of text strings.
```
Comment (manvkaur) [inline azure/functions/decorators/function_app.py:3052]: This one should be about Semantic Search

File: azure/functions/decorators/function_app.py [hunk 3]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+        contains a connection string value.
+        :param collection: The name of the collection or table to search or
+        store.
+        :param query: The semantic query text to use for searching.
+        :param embeddings_model: The ID of the model to use for embeddings.
+        The default value is "text-embedding-ada-002".
+        :param chat_model: The name of the Large Language Model to invoke for
+        chat responses. The default value is "gpt-3.5-turbo".
+        :param system_prompt: Optional The system prompt to use for prompting
+        the large language model.
```
Comment (manvkaur) [inline azure/functions/decorators/function_app.py:3068]: nit: `Optional. The`

File: azure/functions/decorators/function_app.py [hunk 4]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+                                max_chunk_length: Optional[int] = 8 * 1024,
+                                max_overlap: Optional[int] = 128,
+                                data_type: Optional[
+                                    Union[DataType, str]] = None,
+                                **kwargs) \
+            -> Callable[..., Any]:
+        """
+        Supported list of embeddings store is extensible, and more can be
+        added by authoring a specially crafted NuGet package. Visit the
+        currently supported vector specific folder for specific usage
```
Comment (manvkaur) [inline azure/functions/decorators/function_app.py:3117]: any link to be added here?

**Review by manvkaur — APPROVED** [review]
Added some comments related to documentation.

everything else looks good to me.

File: azure/functions/decorators/function_app.py [hunk 5]
```diff
@@ -1435,6 +1443,56 @@ def decorator():
 
         return wrap
 
+    def assistant_skill_trigger(self,
+                                arg_name: str,
+                                task_description: str,
+                                data_type: Optional[
```
Comment (manvkaur) [inline azure/functions/decorators/function_app.py:1449]: AssistantSkillTrigger also supports supplying three more optional parameters - 
image

**Review by manvkaur — CHANGES_REQUESTED** [review]

File: azure/functions/decorators/function_app.py [hunk 6]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+        OpenAI Chat Completions API and return the results to the function.
+
+        Ref: https://platform.openai.com/docs/guides/text-generation/chat-completions-vs-completions  # NoQA
+
+        The examples below define "who is" HTTP-triggered functions with a
+        hardcoded `"who is {name}?"` prompt, where `{name}` is the substituted
+        with the value in the HTTP request path. The OpenAI input binding
+        invokes the OpenAI GPT endpoint to surface the answer to the prompt to
+        the function, which then returns the result text as the response
+        content.
```
[the remark this answers is not in the document]
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py:2826]: Nope. We need to add the examples in the AzureSamples repo

File: azure/functions/decorators/function_app.py (hunk 1 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py]: Fixed

File: azure/functions/decorators/function_app.py (hunk 3 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py]: Fixed

File: azure/functions/decorators/function_app.py (hunk 5 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py:1452]: Added the remaining params

File: azure/functions/decorators/function_app.py (hunk 4 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py:3128]: I intentionally didn't add the link for individual stores. Didn't seem very useful

File: azure/functions/decorators/function_app.py [hunk 7]
```diff
@@ -2740,6 +2790,379 @@ def decorator():
...
+    def assistant_create_output(self, arg_name: str,
+                                data_type: Optional[
+                                    Union[DataType, str]] = None,
+                                **kwargs) \
+            -> Callable[..., Any]:
+        """
+        The assistantCreate output binding creates a new assistant with a
+        specified system prompt.
+
+        :param arg_name: The name of the variable that represents DaprState
```
[the remark this answers is not in the document]
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py]: Fixed

File: azure/functions/decorators/function_app.py (hunk 2 above)
Comment (gavin-aguiar) [reply azure/functions/decorators/function_app.py]: Updated

File: azure/functions/decorators/function_app.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (manvkaur) [reply azure/functions/decorators/function_app.py:2826]: the above text states that The examples below. Should this be updated?

**Review by manvkaur — APPROVED** [review]

**Review by hallvictoria — APPROVED** [review]

**Pull request merged**